x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwkSetLoader.java

```diff
@@ -68,14 +72,18 @@ public JwkSetLoader(final RealmConfig realmConfig, List<String> allowedJwksAlgsP
 
         // Any exception during loading requires closing JwkSetLoader's HTTP client to avoid a thread pool leak
         try {
-            final PlainActionFuture<Tuple<Boolean, JwksAlgs>> future = new PlainActionFuture<>();
+            final PlainActionFuture<JwksAlgs> future = new PlainActionFuture<>();
             reload(future);
             // ASSUME: Blocking read operations are OK during startup
-            final Boolean isUpdated = future.actionGet().v1();
-            assert isUpdated : "initial reload should have updated the JWK set";
-        } catch (Throwable t) {
+            future.actionGet(TimeValue.timeValueMinutes(1));
+        } catch (Exception e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Could not load JWT realm's JWK sets within 1 minute during startup.", e);
+            } else {
+                logger.warn("Could not load JWT realm's JWK sets within 1 minute during startup.");
+            }
             close();
-            throw t;
+            throw e;
```

Comment:
>Is the intention here to ensure node startup will eventually proceed?

The time out is to ensure the startup does not hang (which would be very difficult to troubleshoot without stack trace). The catching Exception instead of Throwable was just general good practice. I agree that we want to avoid failing the node so I will remove the re-throw.  Sorry for the somewhat unrelated change but I needed to change this line of code and noticed this. I can remove from this PR if you think this needs it's own discussion.  (I will also fix s/logger.debug/logger.warn on line 81)